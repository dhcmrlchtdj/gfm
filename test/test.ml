open Mine

let () =
  let cases =
    [
      "`hi`lo`";
      "`foo`";
      "`a  b`";
      "*foo`*`";
      "[not a `link](/foo`)";
      "`<a href=\"`\">`";
      "`<http://foo.bar.`baz>`";
      "*foo bar*";
      "foo*bar*";
      "_foo*";
      "~~foo~~";
      "**foo**";
      "*foo*";
      "<http://foo.bar.baz>";
      "<MAILTO:FOO@BAR.BAZ>";
      "http://commonmark.org";
      "Visit https://encrypted.google.com/search?q=Markup+(business)";
      "p1\n\np2";
      "p1\n---\np2";
      (* https://guides.github.com/features/mastering-markdown/ *)
      "# This is an <h1> tag";
      "###### This is an <h6> tag";
      "*This text will be italic*";
      "_This will also be italic_";
      "**This text will be bold**";
      "__This will also be bold__";
      "_You **can** combine them_";
      "- Item 1\n- Item 2\n    - Item 2a\n\t- Item 2b";
      "![GitHub Logo](/images/logo.png)";
      "Format: ![Alt Text](url)";
      "http://github.com - automatic!";
      "[GitHub](http://github.com)";
      "As Kanye West said:\n\n> We're living the future so\n> the present is our past.";
      "I think you should use an\n`<addr>` element here instead.";
      {|```javascript
function fancyAlert(arg) {
  if(arg) {
    $.facebox({div:'#foo'})
  }
}
```|};
    ]
  in
  let f case = case |> Parser.parse |> Renderer.html_render |> print_endline in
  List.iter f cases
