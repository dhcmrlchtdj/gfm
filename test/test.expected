<p><code>hi</code>lo`</p>
<p><code>foo</code></p>
<p><code>a  b</code></p>
<p>*foo<code>*</code></p>
<p>[not a <code>link](/foo</code>)</p>
<p><code>&lt;a href=&quot;</code>&quot;&gt;`</p>
<p><code>&lt;http://foo.bar.</code>baz&gt;`</p>
<p><em>foo bar</em></p>
<p>foo<em>bar</em></p>
<p>_foo*</p>
<p><del>foo</del></p>
<p><strong>foo</strong></p>
<p><em>foo</em></p>
<p><a rel="noopener" href="http://foo.bar.baz">http://foo.bar.baz</a></p>
<p><a rel="noopener" href="MAILTO:FOO@BAR.BAZ">MAILTO:FOO@BAR.BAZ</a></p>
<p><a rel="noopener" href="http://commonmark.org">http://commonmark.org</a></p>
<p>Visit <a rel="noopener" href="https://encrypted.google.com/search?q=Markup+(business)">https://encrypted.google.com/search?q=Markup+(business)</a></p>
<p>p1</p><br/><p>p2</p>
<p>p1</p><hr/><p>p2</p>
<h1>This is an <a rel="noopener" href="h1">h1</a> tag</h1>
<h6>This is an <a rel="noopener" href="h6">h6</a> tag</h6>
<p><em>This text will be italic</em></p>
<p><em>This will also be italic</em></p>
<p><strong>This text will be bold</strong></p>
<p><strong>This will also be bold</strong></p>
<p><em>You <strong>can</strong> combine them</em></p>
<ul><li>Item 1</li><li>Item 2<ul><li>Item 2a</li><li>Item 2b</li></ul></li></ul>
<p><img src="/images/logo.png" alt="GitHub Logo"/></p>
<p>Format: <img src="url" alt="Alt Text"/></p>
<p><a rel="noopener" href="http://github.com">http://github.com</a> - automatic!</p>
<p><a rel="noopener" href="http://github.com">GitHub</a></p>
<p>As Kanye West said:</p><br/><blockquote><p>We&#x27;re living the future so</p><p>the present is our past.</p></blockquote>
<p>I think you should use an</p><p><code>&lt;addr&gt;</code> element here instead.</p>
<pre><code class="language-javascript">function fancyAlert(arg) {
  if(arg) {
    $.facebox({div:&#x27;#foo&#x27;})
  }
}</code></pre>
